src/oneelement.jl

```diff
@@ -104,19 +153,19 @@ end
     return y
 end
 
-function _mulonel!(y, A, x::OneElementVector, alpha::Number, beta::Number)
+function _mul!(y::AbstractVector, A::AbstractMatrix, x::OneElementVector, alpha, beta)
     check_matmul_sizes(y, A, x)
-    if x.ind[1] ∉ axes(x,1) # in this case x is all zeros
+    if iszero(getindex_value(x))
         mul!(y, A, Zeros{eltype(x)}(axes(x)), alpha, beta)
         return y
     end
```

Comment:
I am dubious as to if this check is worth the potential type instability.
I feel like such a OneElement where that element is also zero is almost never constructed.

Reply:
It's not type-unstable, though. It just calls a different method. This is a guard against the current implementation allowing the indices to not lie within the axes (which probably should be disallowed).